hoptimator-catalog/src/main/java/com/linkedin/hoptimator/catalog/AvroConverter.java

```diff
@@ -74,17 +87,24 @@ public static RelDataType rel(Schema schema, RelDataTypeFactory typeFactory) {
         .filter(x -> x.getValue().getSqlTypeName() != unknown.getSqlTypeName())
         .collect(Collectors.toList()));
     case INT:
-      // schema.isNullable() should be false for basic types iiuc
-      return createRelTypeWithNullability(typeFactory, SqlTypeName.INTEGER, schema.isNullable());
+      return createRelType(typeFactory, SqlTypeName.INTEGER);
     case LONG:
-      return createRelTypeWithNullability(typeFactory, SqlTypeName.BIGINT, schema.isNullable());
+      return createRelType(typeFactory, SqlTypeName.BIGINT);
     case ENUM:
+    case FIXED:
     case STRING:
-      return createRelTypeWithNullability(typeFactory, SqlTypeName.VARCHAR, schema.isNullable());
+      return createRelType(typeFactory, SqlTypeName.VARCHAR);
     case FLOAT:
-      return createRelTypeWithNullability(typeFactory, SqlTypeName.FLOAT, schema.isNullable());
+      return createRelType(typeFactory, SqlTypeName.FLOAT);
     case DOUBLE:
-      return createRelTypeWithNullability(typeFactory, SqlTypeName.DOUBLE, schema.isNullable());
+      return createRelType(typeFactory, SqlTypeName.DOUBLE);
+    case BOOLEAN:
+      return createRelType(typeFactory, SqlTypeName.BOOLEAN);
+    case ARRAY:
+      return typeFactory.createArrayType(rel(schema.getElementType(), typeFactory), -1);
+//  TODO support map types
```

Comment:
any reason we are not enabling this.

Reply:
Need a Calcite version bump first. Map types don't seem to be unparseable in our current version of Calcite. I'll fix the comment. I'll follow up with a version bump at some point.